modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ListTypeDefinition/ListTypeEntriesModal.tsx

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayModal, {useModal} from '@clayui/modal';
+import {
+	API,
+	Input,
+	InputLocalized,
+	invalidateRequired,
+} from '@liferay/object-js-components-web';
+import {openToast} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import {toCamelCase} from '../../utils/string';
+import {ObjectValidationErrors} from './ListTypeFormBase';
+import {fixLocaleKeys} from './utils';
+
+const REQUIRED_MSG = Liferay.Language.get('required');
+const defaultLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
+export interface IModalState extends Partial<PickListItem> {
+	header?: string;
+	itemId?: number;
+	itemKey?: string;
+	modalType?: 'add' | 'edit';
+	pickListId?: number;
+	reloadIframeWindow?: () => void;
+}
+
+function ListTypeEntriesModal() {
+	const [
+		{
+			header,
+			itemId,
+			itemKey,
+			modalType,
+			name_i18n,
+			pickListId,
+			reloadIframeWindow,
+		},
+		setState,
+	] = useState<IModalState>({});
+
+	const [keyChanged, setKeyChanged] = useState(false);
+	const [APIError, setAPIError] = useState<string>('');
+
+	const handleKeyChange = (value: string) => {
+		if (keyChanged === false) {
+			setKeyChanged(true);
+		}
+		setState((previousValues) => ({
+			...previousValues,
+			itemKey: toCamelCase(value),
+		}));
+	};
+
+	const handleNameChange = (name_i18n: LocalizedValue<string>) => {
+		if (
+			modalType !== 'edit' &&
+			keyChanged === false &&
+			name_i18n?.[defaultLanguageId]
+		) {
+			setState((previousValues) => ({
+				...previousValues,
+				itemKey: toCamelCase(name_i18n[defaultLanguageId]!),
+				name_i18n: {...name_i18n},
+			}));
+		}
+		else {
+			setState((previousValues) => ({
+				...previousValues,
+				name_i18n: {...name_i18n},
+			}));
+		}
```

Comment:
Hello, @cgmonte!  To avoid this if-else, could we could do something like this to add the itemKey property using that condition.
```suggestion
		   setState((previousValues) => ({
        ...previousValues,
        ...((modalType !== 'edit' &&
            keyChanged === false &&
            name_i18n?.[defaultLanguageId]) && { itemKey: toCamelCase(name_i18n[defaultLanguageId]!) }),
        name_i18n: { ...name_i18n },
    }));

```

Reply:
Hey, thanks for the suggestion, @alinedoleron! I have updated the code with it.